user_guide.adoc

```diff
@@ -3364,3 +3364,16 @@ seq:
 This is because most processing algorithms require to know size of data
 to process beforehand, and final size of `some_body_type` might be
 determined only in run-time, after parsing took place.
+
+=== Supported stream types
+
+There are different types of streams. There are memory-based streams, file-based streams, and special streams (like sockets pipes etc). Those special streams are often non-seekable and non-tellable.
+
+Some runtime methods use seek() and/or tell() on the stream. If you are parsing memory-streams or file-streams then those operations are fully supported and there is no problem whatsoever. However, if you are using those special types of streams then be aware that some tags are compiled into a code that uses seek and/or tell, and therefore will fail if the stream does not support those. Most tags do not use seek or tell, and do not care what kind of stream it is.
+
+Methods that involve seek and/or tell:
+* expressions using `_io.pos`, `_io.size` and `_io.eof`
+* seq fields using `terminator` with `consume=False`
```

Comment:
Put all the identifiers like `seq`  and `instances` in single back quotes ` to show that they are identifiers, not just words.

Reply:
Fixed.